lib/livebook_web/helpers.ex

```diff
@@ -86,4 +86,30 @@ defmodule LivebookWeb.Helpers do
     pid_str = Phoenix.LiveDashboard.Helpers.encode_pid(pid)
     Routes.live_dashboard_path(socket, :page, node(), "processes", info: pid_str)
   end
+
+  @doc """
+  Converts human-readable strings to strings which can be used
+  as HTML element IDs (compatible with HTML5)
+
+  At the same time duplicate IDs are enumerated to avoid duplicates
+  """
+  @spec names_to_html_ids(Enum.t()) :: Stream.t()
+  def names_to_html_ids(names) do
+    names
+    |> Stream.map(&name_to_html_id/1)
+    |> Stream.transform(%{}, fn html_id, counts ->
+      counts = Map.update(counts, html_id, 1, &(&1 + 1))
+
+      case counts[html_id] do
+        1 -> {[html_id], counts}
+        count -> {["#{html_id}-#{count}"], counts}
+      end
+    end)
```

Comment:
Actually, working with streams here is actually just making things slower. Unless the collections are large, it is probably best to work with lists. And this implementation can likely be done with a single map_reduce pass :)

Reply:
Good to know, thanks!  Changed to use `Enum` all around.

Really the main thing I liked was that `Stream.transform/3` didn't return the accumulator, so the code could be just a bit cleaner.  I used `|> elem(0)` for now...  I could imagine an `Enum.map_reduce_result/3` or something, but then you'd probably need `Enum.flat_map_reduce_result/3` and I don't know if it's worth it 😄